File: 4.semester/ICP/src/multiplier.cpp
/**
* @file multiplier.cpp
* @brief Multiplier implementation
* @author Matúš Liščinský xlisci02, Marcin Vladimír xmarci10
*/
#include "multiplier.h"
#include <iostream>

Multiplier::Multiplier():ArithmBlock(BlockType::mul){}

/**
 * @brief Multiplie two complex numbers
 */
void Multiplier::Execute()
{
    std::vector<std::complex<double>> arr;
    for (Port p : this->inPorts) {
        std::complex<double> c(p.m.at("real"),p.m.at("im"));
        arr.push_back(c);
    }

    std::complex<double> result(1.0,0.0);

    for(auto num: arr){
        result *= num;
    }
    this->outPorts[0].setComplex(result.real(), result.imag());
}

File: 4.semester/IJA/src/application/Multiplier.java
/* 
 * Project:     Implementing Block Editor in JavaFX
 * Course:      IJA 2018
 * Team name:   Tym xzubri00
 * Members: 
 *      Liscinsky Matus xlisci02
 *      Zubrik Tomas    xzubri00
 */
package application;
public class Multiplier extends ArithmBlock{
    
    /**
     * Constructor of Multiplier block, calls super constructor
     */
    public Multiplier() {
    	super("multiplier");
    }
    
    /**
     * Function multiplies 2 complex numbers from input ports and writes result to ouput port
     */
    public void Execute() {
    	Complex num1 = new Complex(this.inPorts.get(0).m.get("real"),this.inPorts.get(0).m.get("im"));
    	Complex num2 = new Complex(this.inPorts.get(1).m.get("real"),this.inPorts.get(1).m.get("im"));
    	
    	Complex result = num1.times(num2);
        this.outPorts.get(0).setComplex(result.re(), result.im());
    }
}
